src/RuleMatchedLog.php

```diff
@@ -261,31 +230,55 @@ public function showItemForm($items_id, $itemtype)
             'START'  => (int)$start,
             'LIMIT'  => (int)$_SESSION['glpilist_limit']
         ];
+        $rows = [];
         foreach ($DB->request($params) as $data) {
-            echo "<tr class='tab_bg_1'>";
-            echo "<td>";
-            echo Html::convDateTime($data['date']);
-            echo "</td>";
-            echo "<td>";
+            $row = [];
+            $row['date'] = Html::convDateTime($data['date']);
             if ($rule->getFromDB($data['rules_id'])) {
-                echo $rule->getLink(1);
+                $row['rulelink'] = $rule->getLink(1);
+            } else {
+                $row[] = '';
             }
-            echo "</td>";
-            echo "<td>";
             if ($agent->getFromDB($data['agents_id'])) {
-                echo $agent->getLink(1);
+                $row['agentlink'] = $agent->getLink(1);
+            } else {
+                $row[] = '';
+            }
+            $row['modulename'] = Request::getModuleName($data['method']);
+            if (isset($data['criteria'])) {
+                $criteria = json_decode($data['criteria'], true);
+                $content = '';
+                $arrayDisplay = function ($array) use (&$arrayDisplay) {
+                    $content = '';
+                    foreach ($array as $key => $value) {
+                        if (is_array($value)) {
+                            $content .= $arrayDisplay($value);
+                        } else {
+                            $content .= '<br>&emsp;&bull;' . $value;
+                        }
+                    }
+                    return $content;
+                };
+                foreach ($criteria as $criterion => $value) {
+                    if (is_array($value)) {
+                        $content .= '<b>' . $criterion . '</b> : ' . $arrayDisplay($value) . '<br>';
+                    } else {
+                        $content .= '<b>' . $criterion . '</b> : ' . $value . '<br>';
+                    }
+                }
+                $row['criteria'] = $content;
+            } else {
+                $row['criteria'] = '';
```

Comment:
This piece of code seems to be here only to format the HTML output. It should be done on template side (you can use a macro to do it).